enterprise/backend/test/metabase_enterprise/audit_app/permissions_test.clj

```diff
@@ -1,26 +1,92 @@
 (ns metabase-enterprise.audit-app.permissions-test
   (:require
+   [clojure.set :as set]
    [clojure.test :refer :all]
+   [metabase-enterprise.audit-app.permissions :as audit-app.permissions]
    [metabase-enterprise.audit-db :as audit-db]
+   [metabase-enterprise.audit-db-test :as audit-db-test]
    [metabase.models.collection :refer [Collection]]
    [metabase.models.collection.graph :refer [update-graph!]]
    [metabase.models.collection.graph-test :refer [graph]]
    [metabase.models.database :refer [Database]]
-   [metabase.models.permissions :refer [Permissions table-query-path]]
+   [metabase.models.permissions
+    :as perms
+    :refer [Permissions table-query-path]]
    [metabase.models.permissions-group :refer [PermissionsGroup]]
    [metabase.models.table :refer [Table]]
-   [metabase.public-settings.premium-features-test :as premium-features-test]
+   [metabase.public-settings.premium-features-test
+    :as premium-features-test]
+   [metabase.query-processor :as qp]
    [metabase.test :as mt]
-   [toucan2.core :as t2]))
+   [metabase.util :as u]
+   [toucan2.core :as t2]
+   [toucan2.tools.with-temp :as t2.with-temp]))
+
+(deftest audit-db-view-names-test
+  (testing "`audit-db-view-names` includes all views in the app DB prefixed with `v_`"
+    (let [view-query "SELECT table_name FROM information_schema.views WHERE table_name LIKE 'v\\_%';"]
+      (is (set/superset?
+           audit-app.permissions/audit-db-view-names
+           (into #{}
+                 (map :table_name (t2/query view-query))))))))
+
+(deftest audit-db-basic-query-test
+  (audit-db-test/with-audit-db-restoration
+   (audit-db/ensure-audit-db-installed!)
```

Comment:
@escherize Do we have any better way to ensure that the audit DB is present in a test? I copied this pattern from other tests but it seems like it always forces the DB to be re-loaded which is pretty slow.

Could we use a macro that just checks whether an audit DB is present, and adds it if not, but otherwise does nothing?

Reply:
We could do that, the reason it's like this is to get a fresh version of it each time, but that might not make a difference anymore.

Reply:
When something about the audit db changes, and you run the test we may get inconsistent results. This way we always test against the same freshly installed one.